be/src/exec/vectorized/hdfs_scanner.cpp

```diff
@@ -165,53 +165,23 @@ uint64_t HdfsScanner::exit_pending_queue() {
     return _pending_queue_sw.reset();
 }
 
-#ifndef BE_TEST
-struct HdfsReadStats {
-    int64_t bytes_total_read = 0;
-    int64_t bytes_read_local = 0;
-    int64_t bytes_read_short_circuit = 0;
-    int64_t bytes_read_dn_cache = 0;
-    int64_t bytes_read_remote = 0;
-};
-
-static void get_hdfs_statistics(hdfsFile file, HdfsReadStats* stats) {
-    struct hdfsReadStatistics* hdfs_stats = nullptr;
-    auto res = hdfsFileGetReadStatistics(file, &hdfs_stats);
-    if (res == 0) {
-        stats->bytes_total_read += hdfs_stats->totalBytesRead;
-        stats->bytes_read_local += hdfs_stats->totalLocalBytesRead;
-        stats->bytes_read_short_circuit += hdfs_stats->totalShortCircuitBytesRead;
-        stats->bytes_read_dn_cache += hdfs_stats->totalZeroCopyBytesRead;
-        stats->bytes_read_remote += hdfs_stats->totalBytesRead - hdfs_stats->totalLocalBytesRead;
-
-        hdfsFileFreeReadStatistics(hdfs_stats);
-    }
-    hdfsFileClearReadStatistics(file);
-}
-#endif
-
 void HdfsScanner::update_counter() {
-#ifndef BE_TEST
-    // _scanner_params.fs is null means scanner open failed
+    static const char* const kHdfsIOProfileSectionPrefix = "HdfsIO";
     if (_scanner_params.fs == nullptr) return;
 
-    HdfsReadStats hdfs_stats;
+    auto res = _scanner_params.fs->get_numeric_statistics();
+    if (!res.ok()) return;
 
-    if (_scanner_params.fs_handle_type == HdfsFsHandle::Type::HDFS) {
-        auto hdfs_file = down_cast<HdfsRandomAccessFile*>(_scanner_params.fs.get())->hdfs_file();
-        if (hdfs_file == nullptr) return;
-        // Hdfslib only supports obtaining statistics of hdfs file system.
-        // For other systems such as s3, calling this function will cause be crash.
-        get_hdfs_statistics(hdfs_file, &hdfs_stats);
-    }
+    std::unique_ptr<NumericStatistics> statistics = std::move(res).value();
+    if (statistics == nullptr || statistics->size() == 0) return;
 
-    auto& root = _scanner_params.parent->_hdfs_io_profile;
-    COUNTER_UPDATE(root.bytes_total_read, hdfs_stats.bytes_total_read);
-    COUNTER_UPDATE(root.bytes_read_local, hdfs_stats.bytes_read_local);
-    COUNTER_UPDATE(root.bytes_read_short_circuit, hdfs_stats.bytes_read_short_circuit);
-    COUNTER_UPDATE(root.bytes_read_dn_cache, hdfs_stats.bytes_read_dn_cache);
-    COUNTER_UPDATE(root.bytes_read_remote, hdfs_stats.bytes_read_remote);
-#endif
+    auto&& profile = _scanner_params.parent->runtime_profile();
+    [[maybe_unused]] auto&& toplev = ADD_TIMER(profile, kHdfsIOProfileSectionPrefix);
```

Comment:
I'm not sure if update_counter will be called more than once in the future.
(e.g. dynmic update profile).
so i suggest move `ADD_TIMER/ADD_CHILD_COUNTER` to other phase.

Reply:
`ADD_TIMER` and `ADD_CHILD_COUNTER` are thread-safe and idempotent, it's OK to call them multiple times.